KyberElasticAdapter/contracts/UniswapV3likeAdapter.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./interfaces/IERC20.sol";
+import "./lib/SafeERC20.sol";
+import "./BaseAdapter.sol";
+
+struct QParams {
+    address tokenIn;
+    address tokenOut;
+    int256 amountIn;
+    uint24 fee;
+}
+
+interface IUniV3Pool {
+    function swap(
+        address recipient,
+        bool zeroForOne,
+        int256 amountSpecified,
+        uint160 sqrtPriceLimitX96,
+        bytes calldata data
+    ) external returns (int256 amount0, int256 amount1);
+
+    function token0() external view returns (address);
+
+    function token1() external view returns (address);
+
+    function liquidity() external view returns (uint128);
```

Comment:
Missing NatSpec
Add descriptions for the interface, events and functions
Refer https://docs.soliditylang.org/en/v0.8.17/natspec-format.html